packit.spec

```diff
@@ -70,35 +39,34 @@ check out packit package for the executable.
 
 %prep
 %autosetup -n %{pypi_name}-%{version}
-# Remove bundled egg-info
-rm -rf %{pypi_name}.egg-info
 
-%if 0%{?rhel}
-# rhbz#1968618 still not fixed for epel-8
-sed -i -e 's|koji|# koji|' setup.cfg
-%endif
+
+%generate_buildrequires
+%pyproject_buildrequires -x testing
+
 
 %build
-%py3_build
+%pyproject_wheel
+
 
 %install
-%py3_install
-python3 setup.py --command-packages=click_man.commands man_pages --target %{buildroot}%{_mandir}/man1
+%pyproject_install
+%pyproject_save_files %{real_name}
+PYTHONPATH="%{buildroot}%{python3_sitelib}" click-man packit --target %{buildroot}%{_mandir}/man1
+
+install -d -m 755 %{buildroot}%{bash_completions_dir}
+cp files/bash-completion/packit %{buildroot}%{bash_completions_dir}/packit
 
-install -d -m 755 %{buildroot}%{_datadir}/bash-completion/completions
-cp files/bash-completion/packit %{buildroot}%{_datadir}/bash-completion/completions/packit
 
 %files
 %license LICENSE
 %{_bindir}/packit
 %{_mandir}/man1/packit*.1*
-%dir %{_datadir}/bash-completion/completions
-%{_datadir}/bash-completion/completions/%{real_name}
+%{bash_completions_dir}/%{real_name}
 
-%files -n python3-%{real_name}
+%files -n python3-%{real_name} -f %{pyproject_files}
```

Comment:
`%{pyproject_files}` already include license.

Reply:
Good to know. `README.md` as well? Why does this not report an error when you install something multiple times?

Reply:
Automatically marking the license files works with Fedora's hatchling version but not with EPEL 9's so better to keep it.

Reply:
Did a test build with and without. For EPEL9, the only non packit package files are the `.dist-info` files:
```
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/INSTALLER
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/METADATA
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/WHEEL
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/entry_points.txt
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/license_files
/usr/lib/python3.9/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/license_files/LICENSE
```
The LICENSE is there, but I presume that is not what we should have right? On the F37 and rawhide I see it is exactly the same:
```
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/INSTALLER
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/METADATA
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/WHEEL
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/entry_points.txt
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/licenses
/usr/lib/python3.11/site-packages/packitos-0.73.0.post11+g293a2ce.d20230412.dist-info/licenses/LICENSE
```

Reply:
You need `rpm -qpL` to show which files are marked with `%license`.

Reply:
> Automatically marking the license files works with Fedora's hatchling version but not with EPEL 9's so better to keep it.

Thanks, that's good to know.

Reply:
@nforro @gotmax23 sorry to ping you over here, but do you guys know how hatchling manages to include the license files there? Recently `scitkit-build-core` adapted a [similar implementation](https://github.com/scikit-build/scikit-build-core/pull/325) to hatchling, writing the license files in the `.dist-info` folder. But for some reason that does not seem to be enough to detect those files as license.
```console
$ rpm -qpL python3-spglib-2.1.0.rc1-1.20230518123112228603.pr272.50.ge38c0dc.x86_64.rpm
warning: python3-spglib-2.1.0.rc1-1.20230518123112228603.pr272.50.ge38c0dc.x86_64.rpm: Header V4 RSA/SHA256 Signature, key ID bcf19e6e: NOKEY
```

Is it the difference in the `METADATA` file:
```
# packit
License-Expression: MIT
License-File: LICENSE
# spglib (where it doesn't detect)
License: BSD-3-Clause
```

Reply:
It seems that way, you need `License-File`: https://src.fedoraproject.org/rpms/pyproject-rpm-macros/blob/9d4e88e1a66fd43ca35ae4eac0f6f70d9227a51b/f/pyproject_save_files.py#_289

Reply:
Oh, lol I was reading that file but for some reason I didn't see that line. Thanks a bunch